app/controllers/security_group_controller.rb

```diff
@@ -191,15 +174,13 @@ def update
   def update_finished
     security_group_id = session[:async][:params][:id]
 
-    if session[:flash_msgs]
-      @flash_array = session[:flash_msgs].dup
-      session[:flash_msgs] = nil
-    end
-
     td = session[:security_group][:task]
     task = MiqTask.find(td[:id])
     if MiqTask.status_ok?(task.status)
-      add_flash("#{td[:resource]} #{td[:action]}d")
+      add_flash(_("%{resource} %{action}d") % {
```

Comment:
There's nothing to translate here. So  i18n is bit extra. WDYT?

Reply:
Ideally, this should be translated, but we would need a list of actions, so that this can be expanded into something like..

```
ACTIONS = {
  :delete => _("%{resource} deleted"),
  :update => _("%{resource} updated"),
  ...
}
add_flash(ACTIONS[td[:action]] % {...
```